### ISSUE TITLE
x/image/bmp: Support writing the alpha channel when present

### ISSUE THREAD

**Issue by kodabb** [golang/go#25945]
### Version/Arch
`go version go1.10.3 darwin/amd64`
issue reproducible with the latest release (on any arch)
```
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
```

### Description
- Encode a image buffer containing an alpha channel using the bmp encoder
- The alpha channel is ignored and an opaque image is produced
- Alpha in BMP seems to be supported in version 4, since Windows 95, according to this spec http://www.fileformat.info/format/bmp/egff.htm

### What did you expect to see?
- I wanted the alpha channel to be preserved, at least when it is present
- Maybe it would be possible to call the `Opaque()` method to see if the image actually contains alpha data to be preserved

### What did you see instead?
- The alpha channel is lost
https://github.com/golang/image/blob/master/bmp/writer.go#L137-L141

**Comment by nigeltao** [comment]
Yeah, it's probably a legitimate feature request, but I don't think I'll have any spare time to look at this soon. Sorry.

**Comment by dwbuiten** [comment]
This can be closed now, I think.
